Xamarin.Forms.Platform.Android.UnitTests/ShellTests.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Android.Content;
+using NUnit.Framework;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.Android.UnitTests;
+
+[assembly: ExportRenderer(typeof(TestShell), typeof(TestShellRenderer))]
+namespace Xamarin.Forms.Platform.Android.UnitTests
+{
+	public class ShellTests : PlatformTestFixture
+	{
+		[Test, Category("Shell")]
+		[Description("Flyout Header Changes When Updated")]
+		public async Task FlyoutHeaderReactsToChanges()
+		{
+			var shell = CreateShell();
+			var initialHeader = new Label() { Text = "Hello" };
+			var newHeader = new Label() { Text = "Hello part 2" };
+			shell.FlyoutHeader = initialHeader;
+			var addedView = await Device.InvokeOnMainThreadAsync(() =>
+			{
+				var r = GetRenderer(shell);
+				var window = (Context.GetActivity()).Window;
+				(window.DecorView as global::Android.Views.ViewGroup).AddView(r.View);
```

Comment:
@hartez the only way I could get shell to work was to add the view to the window. If I don't do this then Fragment inflation doesn't work. Maybe you know a better way to do this? Instead of just adding it to the global window?

Reply:
I don't know a better way off the top of my head.